MoleculeVAE-reImplementation/train.py

```diff
@@ -1,96 +1,330 @@
+
+def frange_cycle_linear(start, stop, n_epoch, n_cycle=4, ratio=0.5):
+    L = np.ones(n_epoch)
+    period = n_epoch / n_cycle
+    step = (stop - start) / (period * ratio)  # linear schedule
+
+    for c in range(n_cycle):
+        v, i = start, 0
+        while v <= stop and (int(i + c * period) < n_epoch):
+            L[int(i + c * period)] = v
+            v += step
+            i += 1
+    return L
+
+
+def frange_cycle_sigmoid(start, stop, n_epoch, n_cycle=4, ratio=0.5):
+    L = np.ones(n_epoch)
+    period = n_epoch / n_cycle
+    step = (stop - start) / (period * ratio)  # step is in [0,1]
+
+    # transform into [-6, 6] for plots: v*12.-6.
+
+    for c in range(n_cycle):
+        v, i = start, 0
+        while v <= stop:
+            L[int(i + c * period)] = 1.0 / (1.0 + np.exp(- (v * 12. - 6.)))
+            v += step
+            i += 1
+    return L
+
+
+#  function  = 1 − cos(a), where a scans from 0 to pi/2
+
+def frange_cycle_cosine(start, stop, n_epoch, n_cycle=4, ratio=0.5):
+    L = np.ones(n_epoch)
+    period = n_epoch / n_cycle
+    step = (stop - start) / (period * ratio)  # step is in [0,1]
+
+    # transform into [0, pi] for plots:
+
+    for c in range(n_cycle):
+        v, i = start, 0
+        while v <= stop:
+            L[int(i + c * period)] = 0.5 - .5 * math.cos(v * math.pi)
+            v += step
+            i += 1
+    return L
+
+
+'''====End of cyclical_annealing module================================'''
+
+'''Read in the training data according to whether the prediction task is performed or not'''
+
+
+def split_validation_dataset(train_dataset, percentage_train):
+    '''
+     This is for building the train and validation datasets
+         - train_dataset:torch.utils.data.Dataset object
+         - percentage_train:percentage of training data default=0.9
+     - Returns:
+        -train_set_final: dataset for training
+        -valid_set_final: dataset for validation
+     '''
+    # use 20% of training data for validation
+    train_set_size = int(len(train_dataset) * percentage_train)
+    valid_set_size = len(train_dataset) - train_set_size
+    # split the train set into two
+    seed = torch.Generator().manual_seed(42)
+    train_set, valid_set = data.random_split(train_dataset, [train_set_size, valid_set_size], generator=seed)
+    train_set_final = torch.utils.data.DataLoader(train_set, batch_size=params['batch_size'])
+    valid_set_final = torch.utils.data.DataLoader(valid_set, batch_size=params['batch_size'])
+
+    return train_set_final, valid_set_final
+
+
+def load_data_by_task(params):
+    '''
+            - Inputs: params (Use hyperparameters.load_params() to load params from expParam.json)
+        - Returns:
+        A pandas DataFrame include follow items:
+            -train_set_final,
+            -valid_set_final,
+            -test_dataset,
+        '''
+    if params['do_prop_pred']:
+        X_train, X_test, Y_train, Y_test, character_index_lookup_dict = load_dataset(params)
```

Comment:
I'm kind of confused here. load_dataset has multiple outputs, but when loading from file the return of load_dataset seems to be a dataframe, so would it then break?

Reply:
Sorry ,`load_dataset ` here maybe a function defined in ` dataloader.py ` file so it should become multiple outputs.

Reply:
Oh I apologize that I found the dataloader.py here is not the updated version!!  I will update them!!sorry!!